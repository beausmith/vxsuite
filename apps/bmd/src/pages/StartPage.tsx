import React, { useContext } from 'react'
import { RouteComponentProps } from 'react-router-dom'

import Article from '../components/Article'
import LinkButton from '../components/LinkButton'

const StartPage = (props: RouteComponentProps) => {
  return (
    <Article>
      <div className="prose">
        <h1>Demo Election</h1>
        <p>
          For each contest, measure, or proposition, select your vote and then
          press the “Next” button.
        </p>
        <h2>Heading 2</h2>
        <p>More text about the election.</p>
        <p>More text about the election.</p>
        <p>
          <LinkButton autoFocus to={`/contests`}>
            Get Started
          </LinkButton>
        </p>
      </div>
    </Article>
  )
}

export default StartPage
