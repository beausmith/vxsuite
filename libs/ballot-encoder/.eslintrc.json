{
  "parserOptions": {
    "project": "./tsconfig.test.json"
  },
  "extends": [
    "plugin:vx/recommended"
  ],
  "rules": {
    "no-bitwise": "off"
  }
}
