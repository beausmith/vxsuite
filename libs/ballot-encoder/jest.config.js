const shared = require('../../jest.config.shared')

module.exports = {
  ...shared,
  setupFilesAfterEnv: ['<rootDir>/test/expect.ts'],
}
