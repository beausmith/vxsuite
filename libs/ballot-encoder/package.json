{
  "name": "@votingworks/ballot-encoder",
  "version": "5.2.1",
  "main": "build/src/index.js",
  "types": "build/src/index.d.ts",
  "files": [
    "build"
  ],
  "scripts": {
    "build": "tsc --build",
    "build:watch": "tsc --build --watch",
    "lint": "eslint .",
    "lint:fix": "eslint . --fix",
    "test": "jest",
    "test:ci": "jest --ci --coverage",
    "test:coverage": "jest --coverage",
    "test:debug": "node --inspect-brk $(which jest) --runInBand --no-cache",
    "test:watch": "jest --watch"
  },
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged"
    }
  },
  "lint-staged": {
    "*.(js|ts)": [
      "eslint --quiet --fix",
      "git add"
    ],
    "*.md": [
      "prettier --write",
      "git add"
    ],
    "package.json": [
      "sort-package-json",
      "git add"
    ]
  },
  "dependencies": {
    "@antongolub/iso8601": "^1.2.1",
    "@votingworks/types": "workspace:*",
    "js-sha256": "^0.9.0",
    "zod": "3.2.0"
  },
  "devDependencies": {
    "@types/jest": "^26.0.15",
    "@types/node": "^14.14.20",
    "@types/text-encoding": "^0.0.35",
    "@typescript-eslint/eslint-plugin": "^4.16.1",
    "@typescript-eslint/parser": "^4.16.1",
    "@votingworks/fixtures": "workspace:*",
    "eslint": "^7.17.0",
    "eslint-config-prettier": "^6.15.0",
    "eslint-import-resolver-node": "^0.3.4",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jest": "^24.1.0",
    "eslint-plugin-no-null": "^1.0.2",
    "eslint-plugin-prettier": "^3.1.4",
    "eslint-plugin-vx": "workspace:*",
    "husky": "^4.3.0",
    "jest": "^26.6.2",
    "jest-watch-typeahead": "^0.6.1",
    "lint-staged": "^10.5.1",
    "prettier": "^2.1.2",
    "random-js": "^2.1.0",
    "sort-package-json": "^1.46.1",
    "ts-jest": "^26.4.3",
    "typescript": "^4.2.0"
  },
  "engines": {
    "node": ">= 12"
  },
  "publishConfig": {
    "access": "public"
  }
}