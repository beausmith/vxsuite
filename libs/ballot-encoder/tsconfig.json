{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "declaration": true,
    "declarationMap": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "isolatedModules": false,
    "lib": ["dom", "dom.iterable", "esnext"],
    "module": "commonjs",
    "moduleResolution": "node",
    "noUncheckedIndexedAccess": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "resolveJsonModule": true,
    "outDir": "build",
    "strict": true,
    "target": "es2015",
    "composite": true
  },
  "include": ["src", "src/data/*.json"],
  "exclude": ["**/*.test.ts", "test"],
  "references": [{ "path": "../fixtures" }, { "path": "../types" }]
}
