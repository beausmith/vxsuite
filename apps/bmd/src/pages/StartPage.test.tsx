import React from 'react'
import { Route } from 'react-router-dom'
import { parseElection } from '@votingworks/ballot-encoder'

import { render } from '../../test/testUtils'
import electionSampleWithSeal from '../data/electionSampleWithSeal.json'
import electionSampleNoSeal from '../data/electionSampleNoSeal.json'
import electionPrimarySample from '../data/electionPrimarySample.json'

import StartPage from './StartPage'

it('renders StartPage', async () => {
  const election = parseElection(electionPrimarySample)
  const { container, getAllByText, getByText } = render(
    <Route path="/" component={StartPage} />,
    {
      ballotStyleId: '12D',
      election,
      precinctId: '23',
      route: '/',
    }
  )
  expect(getAllByText('Democratic Primary Election').length).toBeGreaterThan(1)
  getByText(/ballot style 12D/)
  expect(container.firstChild).toMatchSnapshot()
})

it('renders StartPage with inline SVG', async () => {
  const election = parseElection(electionSampleWithSeal)
  const { container } = render(<Route path="/" component={StartPage} />, {
    election,
    precinctId: '23',
    route: '/',
  })
  expect(container.firstChild).toMatchSnapshot()
})

it('renders StartPage with no seal', async () => {
  const election = parseElection(electionSampleNoSeal)
  const { container } = render(<Route path="/" component={StartPage} />, {
    election,
    precinctId: '23',
    route: '/',
  })
  expect(container.firstChild).toMatchSnapshot()
})
