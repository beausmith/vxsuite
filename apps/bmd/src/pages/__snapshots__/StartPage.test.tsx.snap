// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StartPage 1`] = `
.c1 {
  background: lightgrey;
  border-radius: 0.25rem;
  color: black;
  white-space: nowrap;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
  margin: 1rem;
}

@media print {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    margin: 0;
  }
}

<main
  class="c0"
>
  <div
    class="prose"
  >
    <h1>
      Demo Election
    </h1>
    <p>
      For each contest, measure, or proposition, select your vote and then press the “Next” button.
    </p>
    <h2>
      Heading 2
    </h2>
    <p>
      More text about the election.
    </p>
    <p>
      More text about the election.
    </p>
    <p>
      <button
        class="c1"
      >
        Get Started
      </button>
    </p>
  </div>
</main>
`;
